src/main/java/li/cil/oc2/common/tileentity/NetworkSwitchTileEntity.java

```diff
@@ -0,0 +1,369 @@
+package li.cil.oc2.common.tileentity;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import com.google.gson.internal.LinkedTreeMap;
+import com.mojang.datafixers.util.Pair;
+import li.cil.oc2.api.bus.device.object.Callback;
+import li.cil.oc2.api.bus.device.object.DocumentedDevice;
+import li.cil.oc2.api.bus.device.object.NamedDevice;
+import li.cil.oc2.api.capabilities.NetworkInterface;
+import li.cil.oc2.common.Constants;
+import net.minecraft.block.BlockState;
+import net.minecraft.nbt.*;
+import net.minecraft.tileentity.ITickableTileEntity;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+
+public final class NetworkSwitchTileEntity extends NetworkHubTileEntity implements ITickableTileEntity, NamedDevice, DocumentedDevice {
+    private final String GET_HOST_TABLE = "getHostTable";
+    private final String GET_PORT_CONFIG = "getPortConfig";
+    private final String SET_PORT_CONFIG = "setPortConfig";
+    private final String GET_LINK_STATE = "getLinkState";
+
+    // Keep Mac Address entries for 2 Real Life Minutes
+    private final long HOST_TTL = 20 * 60 * 2;
+    private final Map<Long, HostEntry> hostTable = new HashMap<>();
```

Comment:
I'd probably make this a `Long2ObjectMap` to avoid the boxing of the key; I can do that after the merge though. `Long2ObjectArrayMap` would probably be sufficient with the number of devices I'd expect on such a network?

Reply:
I am not quite sure on the performance characteristics, but i think a realistic number would be < 50. Fairly sure thats good enough.